ajax cache fixes

narzac opened the pull request:
## What I did

1. Race condition fix for on the fly requests
2. Refactor code, cleanup

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/ing-bank/lion?pullRequest=1385) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/ing-bank/lion?pullRequest=1385) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: ruphin<br/>:white_check_mark: narzac<br/>:x: Martin Pool<br/><hr/>**Martin Pool** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/ing-bank/lion?pullRequest=1385) it.</sub>

ruphin reviewed (approved):
LGTM

gerjanvangeest reviewed (approved):
LGTM

narzac commented:
Fix cache session race condition for in-flight requests

**BREAKING** public API changes:
  - Changed `timeToLive` to `maxAge`
  - Renamed `requestIdentificationFn` to `requestIdFunction`
